### ISSUE TITLE
add badges for youth savings program

### ISSUE THREAD

**Issue by elasticspoon** [rubyforgood/casa#5754]
## What type(s) of User Does This Feature Affect?

 - supervisors
 - admins
 - volunteers

## Description

Part of #3027
Blocked by #5753

Youth 14+ in foster care in MD are supposed to receive money deposited into a “Foster Youth Savings” account each year. 

We have created basic CRUD functionality to modify and display the state of a case savings, now we want to enhance it with badges to make is simpler to track the status of the deposits on a per case basis.

Consider the emancipation button as an example 

![Image](https://github.com/rubyforgood/casa/assets/14540596/ceb844e5-9c12-4922-978b-f88cf1655e91)

We would like to see this data on the case itself, as well as most pages that show cases within tables (`/casa_cases`, `/volunteers`).

![Image](https://github.com/rubyforgood/casa/assets/14540596/c1124af4-317e-4e91-a978-c12ddcbb80d4)

### Criteria for Completion

- [ ] Create a simple and minimal component that will show the status of savings deposit for a particular case for the current year. We care about annual values and one time values only if they are still available.
- [ ] Consider how to link to the index page that shows all the savings values
- [ ] Consider how to display more data with the badge. Should it be a dropdown that expands to give more details?
- [ ] Add the component to the case page `/casa_cases/:case_id`
- [ ] Add the component to the cases page `/casa_cases`
- [ ] Add the component to the volunteers page `/volunteers`

## How to Access the QA Site

_Login Details:_
[Link to QA site](https://casa-qa.herokuapp.com/)

Login Emails:
- volunteer1@example.com view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`

password for all users: 12345678

### Questions? Join Slack!

We highly recommend that you join us in [slack](https://join.slack.com/t/rubyforgood/shared_invite/zt-21pyz2ab8-H6JgQfGGI0Ab6MfNOZRIQA) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

**Comment by compwron** [comment]
Closing for inactivity — this issue has been open for a long time with no recent activity, and is most likely no longer relevant to the current state of the project (or has been overtaken by events).

If this still matters to you or has resurfaced, please feel free to:
- **Reopen** with a quick comment on why it's still needed, or
- **Open a fresh issue** describing the current behavior and what you'd like to see — much easier than picking up multi-year-old context.

Thanks for understanding 💜
